rust/lance/src/dataset/write/update.rs

```diff
@@ -1048,4 +1069,273 @@ mod tests {
+            .iter()
+            .find(|idx| idx.name == "vec_idx")
+            .unwrap();
+
+        let str_bitmap = updated_str_index.fragment_bitmap.as_ref().unwrap();
+        assert_eq!(str_bitmap.len(), 3);
+        assert_eq!(str_bitmap.iter().collect::<Vec<_>>(), vec![0, 1, 2]);
+
+        let vec_bitmap = updated_vec_index.fragment_bitmap.as_ref().unwrap();
+        assert_eq!(vec_bitmap.len(), 2);
+        assert_eq!(vec_bitmap.iter().collect::<Vec<_>>(), vec![0, 1]);
+
+        let fragments = updated_dataset.get_fragments();
+        assert!(fragments.len() > 2);
+
+        let second_fragment = &fragments[1];
+        assert!(second_fragment
+            .get_deletion_vector()
+            .await
+            .unwrap()
+            .is_some());
+    }
+
+    #[tokio::test]
+    async fn test_update_mixed_indexed_unindexed_fragments() {
+        let mut dataset = lance_datagen::gen_batch()
+            .col(
+                "str",
+                lance_datagen::array::cycle_utf8_literals(&["a", "b", "c", "d", "e", "f"]),
+            )
+            .col(
+                "vec",
+                lance_datagen::array::rand_vec::<Float32Type>(Dimension::from(4)),
+            )
+            .into_ram_dataset_with_params(
+                FragmentCount::from(2),
+                FragmentRowCount::from(3),
+                Some(WriteParams {
+                    max_rows_per_file: 3,
+                    enable_stable_row_ids: true,
+                    ..Default::default()
+                }),
+            )
+            .await
+            .unwrap();
+
+        dataset
+            .create_index(
+                &["str"],
+                IndexType::Scalar,
+                Some("str_idx".to_string()),
+                &ScalarIndexParams::default(),
+                true,
+            )
+            .await
+            .unwrap();
+
+        dataset
+            .create_index(
+                &["vec"],
+                IndexType::Vector,
+                Some("vec_idx".to_string()),
+                &VectorIndexParams::ivf_flat(1, MetricType::L2),
+                true,
+            )
+            .await
+            .unwrap();
+
+        let initial_indices = dataset.load_indices().await.unwrap();
+        let str_index = initial_indices
+            .iter()
+            .find(|idx| idx.name == "str_idx")
+            .unwrap();
+        let vec_index = initial_indices
+            .iter()
+            .find(|idx| idx.name == "vec_idx")
+            .unwrap();
+
+        assert_eq!(str_index.fragment_bitmap.as_ref().unwrap().len(), 2);
+        assert!(str_index.fragment_bitmap.as_ref().unwrap().contains(0));
+        assert!(str_index.fragment_bitmap.as_ref().unwrap().contains(1));
+        assert_eq!(vec_index.fragment_bitmap.as_ref().unwrap().len(), 2);
+        assert!(vec_index.fragment_bitmap.as_ref().unwrap().contains(0));
+        assert!(vec_index.fragment_bitmap.as_ref().unwrap().contains(1));
```

Comment:
Instead of having to make all these assertions, maybe it would just be easier to directly assert equality:
```suggestion
        assert_eq!(str_index.fragment_bitmap.as_ref(), &RoaringBitmap::from([0, 1]));
        assert_eq!(vec_index.fragment_bitmap.as_ref(), &RoaringBitmap::from([0, 1]));
```